railties/lib/rails/generators/rails/benchmark/templates/benchmark.rb.tt

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+require_relative "../config/environment"
+
```

Comment:
In my experience, benchmarks tend to do some setup here that they act on in the reports. Wonder if we should at least note that with a comment. Or maybe there's a way to set up some sample data that's not annoying ala `DATA = 100.times.map { "example" }`. Or:

```ruby
# Override the method under test and optimize it.
class Post
  def optimized_title
    title
  end
end

# Find a representative sample.
post = Post.first

Benchmark.ips do |x|
  x.report("before") { post.title }
  x.report("after")   { post.optimized_title }
end
```

Hell, we could consider adding a `require "active_record/test_case/benchmark_helper` then if people ran this in the `test` environment, we'd automatically wrap the script in a transaction that would be rolled back after the benchmark. Just some more ideas for what we could do since we're automating things here.

Reply:
🤔 I'm good with leaving a comment in the template and leave the decision with the end user.

Unless there is a general setup that works best for _setting_ things up?

The benchmark_helper is an interesting idea, although is it maybe too prescriptive?

-----

I'm 100% open to suggestions on how to move forward with this comment.

Reply:
> Unless there is a general setup that works best for _setting_ things up?

That's benchmark dependent, I guess. But it's stuff like this that would be good to think about such that we can make calls on it. What would people be testing here? Are these one-off scripts and what's under test? Active Record interactions? Full view tests? (Probably not). Seems like this is at the model level, so I think it's fine to make model related decisions, while still leaving it possible to remove that stuff for other testing.

Reply:
I like this idea.

Common benchmark mistake:

```ruby 
x.report("thing") { MyService.perform(User.first) }
```

Reifying a "setup method" or "setup area" where data is prepared sounds good to me.

Reply:
I assume we would do this with a comment and some whitespace above the benchmark?